### PR TITLE
Add phpstan and psalm

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

### CODE REVIEW

**Comment by jordisala1991** [comment]
Hum... not sure how to fix that one... it is suposed to understand that I add the getId method on the mock creation

**Comment by VincentLanglet** [comment]
> Hum... not sure how to fix that one... it is suposed to understand that I add the getId method on the mock creation

Similar to https://github.com/sonata-project/SonataAdminBundle/blob/master/phpstan-baseline.neon#L3-L5

**Comment by jordisala1991** [comment]
> > Hum... not sure how to fix that one... it is suposed to understand that I add the getId method on the mock creation
> 
> Similar to https://github.com/sonata-project/SonataAdminBundle/blob/master/phpstan-baseline.neon#L3-L5

Trying that solution right now, the best thing is that on my env I do not get this error. Let's see (anyway on master it does not need the supression because the error does not exist)

**Comment by jordisala1991** [comment]
Nice , I can't generate locally the baseline and I don't know why it is not matching..

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -1,6 +1,18 @@
 parameters:
     ignoreErrors:
-        - # NEXT_MAJOR: Remove this issue.
+        - # NEXT_MAJOR: Remove this issue
+            message: "#^Trying to mock an undefined method [a-zA-Z]*\(\) on class Sonata\\\\ClassificationBundle\\\\Model\\\\CategoryInterface\\.$#"
```
Comment (VincentLanglet) [inline phpstan-baseline.neon:4]: Sonata\\ClassificationBundle\\Model\\CategoryInterface\.

Comment (jordisala1991) [reply phpstan-baseline.neon]: Fixed.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**